File: library/src/main/java/com/iktwo/binder/ListDelegateAdapter.java
package com.iktwo.binder;

import android.support.annotation.NonNull;

import java.util.Collections;
import java.util.List;

public class ListDelegateAdapter extends DelegateAdapter {
    private List<?> items;

    public ListDelegateAdapter(@NonNull List<?> items) {
        this.items = items;
    }

    public ListDelegateAdapter() {
        this.items = Collections.emptyList();
    }

    public void setItems(List<?> items) {
        this.items = items;
        notifyDataSetChanged();
    }

    @Override
    public Object getItem(int position) {
        return items.get(position);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }
}

File: library/src/main/java/com/iktwo/binder/ListDelegateAdapter.kt
package com.iktwo.binder

class ListDelegateAdapter<T : Any>(items: List<T> = emptyList()) : DelegateAdapter() {
    private var items: List<T>? = items

    fun setItems(items: List<T>) {
        this.items = items
        notifyDataSetChanged()
    }

    override fun getItem(position: Int): Any {
        return items!![position]
    }

    override fun getItemCount(): Int {
        return items!!.size
    }
}
